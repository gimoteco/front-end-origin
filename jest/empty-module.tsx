import * as React from 'react';
const SvgrMock = () => <span />;

export default SvgrMock;

export const ReactComponent = SvgrMock;
