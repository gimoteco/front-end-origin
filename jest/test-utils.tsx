import { render } from '@testing-library/react';
import * as React from 'react';
import { ThemeProvider } from 'styled-components';
import theme from '../src/styles/theme';

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (ui: React.ReactElement, options = {}) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from '@testing-library/react';
// override render method
export { customRender as render };
