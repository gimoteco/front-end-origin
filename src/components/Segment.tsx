import * as React from 'react';
import * as Style from './Segment.styles';

interface SegmentProps {
  children: {
    top: React.ReactNode;
    bottom: React.ReactNode;
  };
}

function Segment({ children: { top, bottom } }: SegmentProps) {
  return (
    <Style.SegmentWrapper>
      <Style.UpWrapper>{top}</Style.UpWrapper>
      <Style.DownWrapper>{bottom}</Style.DownWrapper>
    </Style.SegmentWrapper>
  );
}

export default React.memo(Segment);
