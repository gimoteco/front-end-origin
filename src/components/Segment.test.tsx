import React from 'react';
import { render } from '../../jest/test-utils';
import Segment from './Segment';

describe('Segment', () => {
  it('should render according snapshot', () => {
    const { container } = render(
      <Segment>{{ top: 'Hello World', bottom: 'an awesome world' }}</Segment>
    );

    expect(container).toMatchSnapshot();
  });
});
