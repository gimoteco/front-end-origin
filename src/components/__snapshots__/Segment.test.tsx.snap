// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Segment should render according snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa egBlud"
  >
    <div
      class="sc-bwzfXH UBTUY"
    >
      Hello World
    </div>
    <div
      class="sc-htpNat fSDQmJ"
    >
      an awesome world
    </div>
  </div>
</div>
`;
